From: Ian Kent <ikent@redhat.com>
Date: Tue, 12 Aug 2008 12:25:43 +0800
Subject: [autofs4] fix mntput, dput order bug
Message-id: 20080812042541.15842.61624.stgit@web.messagingengine.com
O-Subject: [RHEL 5.3 PATCH 04/20] autofs4 - fix mntput, dput order bug [Realted: bz452139]
Bugzilla: 452139

This patch is an upstream bug fix and shoud be associated with
bug 452139.

Fix a bug in autofs4_mount_busy() where mntput() was called before dput().
This is a hunk taken from an LKML submission by Jan Blunck <jblunck@suse.de>.

diff --git a/fs/autofs4/expire.c b/fs/autofs4/expire.c
index cfa12db..894fee5 100644
--- a/fs/autofs4/expire.c
+++ b/fs/autofs4/expire.c
@@ -73,8 +73,8 @@ static int autofs4_mount_busy(struct vfsmount *mnt, struct dentry *dentry)
 	status = 0;
 done:
 	DPRINTK("returning = %d", status);
-	mntput(mnt);
 	dput(dentry);
+	mntput(mnt);
 	return status;
 }
 
